Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -87,13 +87,16 @@ public SiteExtensionManager(IContinuousJobsManager continuousJobManager, ITrigge
                             .OrderByDescending(p => p.DownloadCount);
             }
 
-            using (tracer.Step("Convert search result to SiteExtensionInfos"))
+            using (tracer.Step("Convert search result to SiteExtensionInfos with max concurrent requests: {0}", System.Environment.ProcessorCount))
             {
+                // GetResourceAsync is not thread safe
```

Comment:
public static async Task<T> GetResourceAndValidateAsync<T>(this SourceRepository srcRepo) where T : class, INuGetResource
        {
            var resource = await srcRepo.GetResourceAsync<T>();

GetResourceAsync is the nuget method being called in the end